packages/page-staking/src/Suspensions/Suspensions.tsx

```diff
@@ -92,6 +85,16 @@ function parseEvents (events: EventRecord[], productionBanConfigPeriod: number,
     }).flat();
 }
 
+function updateLiftEraForBansLiftedManually (eventsInBlocks: SuspensionEvent[], banned: Banned[], sessionInfo: SessionInfo) {
+  eventsInBlocks.forEach((eventInBlock) => {
+    if (banned.find((banned) => banned.account === eventInBlock.address) === undefined) {
+      if (eventInBlock.suspensionLiftsInEra > sessionInfo.currentEra) {
+        eventInBlock.suspensionLiftsInEra = sessionInfo.currentEra;
+      }
+    }
+  });
+}
```

Comment:
I think it would be better to create new objects instead of mutating them. This way it's simpler to reason about the changes and the state updates are reactive, something like this:

```suggestion
function updateLiftEraForBansLiftedManually (eventsInBlocks: SuspensionEvent[], banned: Banned[], sessionInfo: SessionInfo) {
  return eventsInBlocks.map((eventInBlock) => {
    if (
      banned.some((banned) => banned.account === eventInBlock.address) ||
      eventInBlock.suspensionLiftsInEra <= sessionInfo.currentEra
    ) {
      return eventInBlock;
    }
    
    return { ...eventInBlock, suspensionLiftsInEra: sessionInfo.currentEra };
  });
}
```

Reply:
Yeah good point, will do. This is a habit from backend programming when mutating object in big big array is faster than copying whole array with one changed object, but you are definitely right the readiness and code safety is better concern here than some optimization.